### PR TITLE
refactor(material-experimental/mdc-chips): implement trailing icon foundation

### PR DESCRIPTION
**Pull request by crisbeto**
The new foundation isn't being used yet, but these changes implement it so it's easier to roll out in the future.

### CODE REVIEW

File: src/material-experimental/mdc-chips/chip-icons.ts [hunk 1]
```diff
@@ -57,8 +59,40 @@ export class MatChipAvatar {
...
-  constructor(public _elementRef: ElementRef) {}
+export class MatChipTrailingIcon implements OnDestroy {
+  private _foundation: MDCChipTrailingActionFoundation;
+  private _adapter: MDCChipTrailingActionAdapter = {
+    focus: () => this._elementRef.nativeElement.focus(),
+    getAttribute: (name: string) => this._elementRef.nativeElement.getAttribute(name),
+    setAttribute: (name: string, value: string) => {
+      this._elementRef.nativeElement.setAttribute(name, value);
+    },
+    // TODO(crisbeto): this also a `trigger` parameter that the chip isn't
```
Comment (jelbourn) [inline src/material-experimental/mdc-chips/chip-icons.ts]: "this also" -> "there's also" ?

File: src/material-experimental/mdc-chips/chip-icons.ts [hunk 2]
```diff
@@ -57,8 +59,40 @@ export class MatChipAvatar {
...
+    // handling yet. Consider passing it along once MDC start using it.
+    notifyInteraction: () => {
+      // this._chip._notifyInteraction();
```
Comment (jelbourn) [inline src/material-experimental/mdc-chips/chip-icons.ts]: Could we replace the commented code with a TODO comment?

**Comment by crisbeto** [comment]
I've addressed the feedback.

**Comment by crisbeto** [comment]
Closing in favor of https://github.com/angular/components/pull/19318.

**Pull request closed without merging**